spec/vra_spec.rb

```diff
@@ -102,14 +103,34 @@
       expect(state[:resource_id]).to eq('e8706351-cf4c-4c12-acb7-c90cc683b22c')
     end
 
-    describe 'getting the IP address from the server' do
+    describe 'getting the IP address from the server when use_dns is false' do
```

Comment:
We should refactor these test describes/contexts to be a bit more clear now that we have a context to switch on.

Proposed format:

```ruby
describe 'setting the hostname in the state hash' do
  context 'when use_dns is true' do
    it 'raises an exception if the server name is nil' do
      # test criteria here
    end
    it 'uses the server name as the hostname' do
      # test criteria here
    end
  end
  context 'when use_dns is false' do
    it 'raises an exception if no IP address is available' do
      # test criteria here
    end
    it 'uses the IP address as the hostname' do
      # test criteria here
    end
  end
end
```

Can you take a stab at that refactor?